crypto/dsa/dsa_ameth.c

```diff
@@ -63,6 +65,7 @@ static int dsa_pub_decode(EVP_PKEY *pkey, X509_PUBKEY *pubkey)
         goto err;
     }
 
+    dsa->dirty_cnt++;
```

Comment:
If more than one thread touches this at the same time, it will be changed either way, right?  At least one increment will happen, and that is enough to make this structure "dirty", because the key transferring mechanism looks for dirty count equality only.

Reply:
Good point.